### PR TITLE
Us1/feature/when new customer

### PR DESCRIPTION
**Pull request by alexsorindrago**
test working

### CODE REVIEW

File: src/main/java/com/bucur/model/Customer.java [hunk 1]
```diff
@@ -13,6 +13,12 @@
     private String emailAddress;
 
     public Customer() {
+        withId(id);
```
Comment (cosminbucur) [inline src/main/java/com/bucur/model/Customer.java]: this is not a valid build pattern implementation. please read here: http://www.javabyexamples.com/builder-vs-fluent-interface/

File: src/main/java/com/bucur/service/MountingService.java [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+import java.util.List;
 
 public class MountingService {
 
-    public void replaceTyres(CustomerVisit customerVisit, long mechanicId, String licensePlate, Vehicle oldVehicleTyres, Vehicle newVehicleTyres) {
+    private List<Tyre> newTyres = new ArrayList<>();
+
+    public void replaceTyres(CustomerVisit customerVisit, long mechanicId, String licensePlate, Vehicle vehicleWithOldTyres, Vehicle vehicleWithNewTyres) {
+
+        Mechanic.mechanic().setId(mechanicId);
```
Comment (cosminbucur) [inline src/main/java/com/bucur/service/MountingService.java]: if you read this line of code, it doesn't make much sense in English. so something must be wrong.

File: src/main/java/com/bucur/service/MountingService.java [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
 
 public class MountingService {
 
-    public void replaceTyres(CustomerVisit customerVisit, long mechanicId, String licensePlate, Vehicle oldVehicleTyres, Vehicle newVehicleTyres) {
+    private List<Tyre> newTyres = new ArrayList<>();
+
+    public void replaceTyres(CustomerVisit customerVisit, long mechanicId, String licensePlate, Vehicle vehicleWithOldTyres, Vehicle vehicleWithNewTyres) {
```
Comment (cosminbucur) [inline src/main/java/com/bucur/service/MountingService.java]: are the tests still passing using this implementation?

Comment (alexsorindrago) [reply src/main/java/com/bucur/service/MountingService.java]: yes, all tests are working at the moment.

File: src/main/java/com/bucur/service/MountingService.java (hunk 2 above)
Comment (alexsorindrago) [reply src/main/java/com/bucur/service/MountingService.java]: looking into it now

File: src/main/java/com/bucur/model/Customer.java (hunk 1 above)
Comment (alexsorindrago) [reply src/main/java/com/bucur/model/Customer.java]: thanks. looking into it.

File: src/main/java/com/bucur/dto/Vehicle.java [hunk 4]
```diff
@@ -11,6 +13,14 @@
     public Vehicle() {
     }
 
+    public void mountTyres(List<Tyre> newTyres) {
```
[the remark this answers is not in the document]
Comment (alexsorindrago) [reply src/main/java/com/bucur/dto/Vehicle.java]: thanks. looking into it.

**Pull request merged**